src/common/signalk/SKNMEA2000Parser.cpp

```diff
@@ -99,23 +149,155 @@ const SKUpdate& SKNMEA2000Parser::parse128267(const SKSourceInput& input, const
+  tN2kRudderDirectionOrder rudderDirectionOrder;
+  double rudderPosition;
+  double angleOrder;
+
+  if (ParseN2kRudder(msg,rudderPosition,instance,rudderDirectionOrder,angleOrder) ) {
+    SKUpdateStatic<1> *update = new SKUpdateStatic<1>();
+    update->setTimestamp(timestamp);
+
+    SKSource source = SKSource::sourceForNMEA2000(input, msg.PGN, msg.Priority, msg.Source);
+    update->setSource(source);
+
+    if (!N2kIsNA(rudderPosition) && (rudderPosition < M_PI_2 / 2) && (rudderPosition > M_PI_2 / 2) ) {
+      // -> Current rudder angle, +ve is rudder to Starboard
+      update->setSteeringRudderAngle(rudderPosition);
+      DEBUG("Rudder: %d",SKRadToDeg(rudderPosition));
+    }
+
+    _sku = update;
+    return *_sku;
+  }
+  else {
+    DEBUG("Unable to parse NMEA2000 with PGN %i", msg.PGN);
+    return _invalidSku;
+  }
+}
+// *****************************************************************************
+//    PGN 127250 VESSEL HEADING RAPID
+//    Heading sensor value with a flag for True or Magnetic.
+//    Heading               Heading in radians
+//      - Deviation         Magnetic deviation in radians. Use N2kDoubleNA for undefined value.
+//      - Variation         Magnetic variation in radians. Use N2kDoubleNA for undefined value.
+//      tN2kHeadingReference
+//        N2khr_true=0,
+//        N2khr_magnetic=1
+// *****************************************************************************
+const SKUpdate& SKNMEA2000Parser::parse127250(const SKSourceInput& input, const tN2kMsg& msg, const SKTime& timestamp) {
+  unsigned char sid;
+  tN2kHeadingReference headingReference;
+  double heading = N2kDoubleNA;
+  double deviation = N2kDoubleNA;
+  double variation = N2kDoubleNA;
+
+  if (ParseN2kHeading(msg,sid,heading,deviation,variation,headingReference) ) {
+
+    if (headingReference == N2khr_magnetic) {
+      // Deviation and Variation could be sent
+      // (Timo's NMEA Simulator is sending Dev and Var at magnetic HDG only)
+      heading += deviation + variation;
+      //TODO put 3 when updated to deviation
+      SKUpdateStatic<2> *update = new SKUpdateStatic<2>();
+      update->setTimestamp(timestamp);
+
+      SKSource source = SKSource::sourceForNMEA2000(input, msg.PGN, msg.Priority, msg.Source);
+      update->setSource(source);
+      update->setNavigationHeadingMagnetic(heading);
+      if (!N2kIsNA(variation))
+        update->setNavigationMagneticVariation(variation);
+        /* coming when Signal K adds deviation
+        if (!N2kIsNA(deviation))
+          update->setNavigationMagneticDeviation(deviation);
+        */
+      _sku = update;
+      return *_sku;
+    }
+    if ( headingReference == N2khr_true ) {
+      SKUpdateStatic<1> *update = new SKUpdateStatic<1>();
+      update->setTimestamp(timestamp);
+
+      SKSource source = SKSource::sourceForNMEA2000(input, msg.PGN, msg.Priority, msg.Source);
+      update->setSource(source);
+      update->setNavigationHeadingTrue(heading);
```

Comment:
Are you sure that headingTrue never includes information on the deviation? Would be useful for a client to calculate the magnetic heading. I think we could support this with very little extra effort.

Reply:
I think I remember, that I have looked at Timos NMEA Simulator to find out that it is just sending with the Mag Hdg.

Pros and Cons:
Yes, it could/should be done with a config-setting if someone wants it or not.
No, IMHO Heading and Depth have to be treated very carefully and should never be changed or calculated if the user doesn't know it.

Reply:
But, we could also translate PGN127258 Magnetic Variation which might come from GPS/Plotter